pkg/console/controllers/managedclusters/controller.go

```diff
@@ -0,0 +1,450 @@
+			continue
+		}
+
+		// Skip if unable to get managed cluster API server config map for any other reason
+		if err != nil {
+			klog.V(4).Infof("Error getting CA file for managed cluster %v", clusterName)
+			continue
+		}
+
+		managedClusterConfigs = append(managedClusterConfigs, consoleserver.ManagedClusterConfig{
+			Name: clusterName,
+			APIServer: consoleserver.ManagedClusterAPIServerConfig{
+				URL:    clientConfig.URL,
+				CAFile: fmt.Sprintf("%s/%s/%s", api.ManagedClusterAPIServerCAMountDir, configmapsub.APIServerCAConfigMapName(clusterName), api.ManagedClusterAPIServerCAKey),
+			},
+		})
+	}
+
+	if len(managedClusterConfigs) > 0 {
+		required, err := configmapsub.DefaultManagedClustersConfigMap(operatorConfig, managedClusterConfigs)
+		if err != nil {
+			return err, "FailedMarshallingYAML"
+		}
+
+		if _, _, applyErr := resourceapply.ApplyConfigMap(c.configMapClient, recorder, required); applyErr != nil {
+			return applyErr, "FailedApply"
+		}
+	}
+
+	return nil, ""
+}
+
+func (c *ManagedClusterController) SyncManagedClusterViewOAuthClient(ctx context.Context, operatorConfig *operatorv1.Console) ([]*unstructured.Unstructured, error, string) {
+	managedClusters, listErr := c.listManagedClusters(ctx)
+	if listErr != nil || len(managedClusters.Items) == 0 {
+		return nil, listErr, "FailedList"
+	}
+
+	errs := []string{}
+	managedClusterOAuthClientViews := []*unstructured.Unstructured{}
+	for _, managedCluster := range managedClusters.Items {
+		mcvOAuth := managedclusterviewsub.DefaultViewOAuthClient(operatorConfig, managedCluster.Name)
+		gvr := managedclusterviewsub.GetGVR()
+
+		oAuthResp, oAuthErr := c.dynamicClient.Resource(gvr).Namespace(managedCluster.Name).Create(ctx, mcvOAuth, metav1.CreateOptions{})
+		if oAuthErr != nil && apierrors.IsAlreadyExists(oAuthErr) {
+			mcvOAuthName, _ := managedclusterviewsub.GetName(mcvOAuth)
+			oAuthResp, oAuthErr = c.dynamicClient.Resource(gvr).Namespace(managedCluster.Name).Get(ctx, mcvOAuthName, metav1.GetOptions{})
+		}
+
+		if oAuthErr != nil || oAuthResp == nil {
+			errs = append(errs, fmt.Sprintf("Error syncing managed cluster view for oauth client for cluster %s: %v", managedCluster.Name, oAuthErr))
+		} else {
+			managedClusterOAuthClientViews = append(managedClusterOAuthClientViews, oAuthResp)
+		}
+	}
+
+	if len(errs) > 0 {
+		return nil, errors.New(strings.Join(errs, "\n")), "ManagedClusterViewOAuthClientSyncError"
+	}
+
+	return managedClusterOAuthClientViews, nil, ""
+}
+
+func (c *ManagedClusterController) SyncManagedClusterActionCreateOAuthClient(ctx context.Context, operatorConfig *operatorv1.Console, managedClusterOAuthClientViews []*unstructured.Unstructured) ([]*unstructured.Unstructured, error, string) {
+	managedClusterList := []string{}
+	managedClusterListErrors := []string{}
+	for _, managedClusterOAuthView := range managedClusterOAuthClientViews {
+		status, statusErr := managedclusterviewsub.GetStatus(managedClusterOAuthView)
+		if !status || statusErr != nil {
```

Comment:
If I'm reading correctly, we create an MCA for any MCV where we don't know a status, but we do know a namespace? Seems like there could be a race condition here where the status hasn't had time to propagate to the MCV and we try to create an MCA.